test/test_predict_command.py

```diff
@@ -34,25 +35,119 @@ def setup_module():
-        print(result)
-        assert not result.isnull().any().any()
+            fd.close()  # Explicitly close
+            
+        with tempfile.NamedTemporaryFile(delete=False, suffix=".csv") as fd_out:
+            deletes.append(fd_out.name)
+            full_args = [fd.name] + args + ["--out", fd_out.name]
+            print("Running with args: %s" % full_args)
+            predict_command.run(full_args)
+            fd_out.close()  # Explicitly close
+            result = pandas.read_csv(fd_out.name)
+            print(result)
+            assert not result.isnull().any().any()
     finally:
+        # Make sure we've closed the files and pandas has released them
+        import gc
+        gc.collect()  # Force garbage collection
+        
+        # Add a small delay to ensure files are released
+        import time
+        time.sleep(0.1)
+        
         for delete in deletes:
-            os.unlink(delete)
+            try:
+                os.unlink(delete)
+            except Exception as e:
+                print(f"Warning: Could not delete {delete}: {e}")
 
     assert result.shape == (3, 8)
 
 
+def test_tensorflow_vs_pytorch_backends():
+    """Test that tensorflow and pytorch backends produce matching results."""
+    args = [
+        "--alleles", "HLA-A0201",
+        "--alleles", "HLA-A0201", "HLA-A0301",
+        "--peptides", "SIINFEKL", "SIINFEKD", "SIINFEKQ",
+        "--prediction-column-prefix", "mhcflurry_",
+        "--affinity-only",
+    ]
+
+    deletes = []
+    result_tf = None
+    result_torch = None
+    
+    try:
+        # Run with tensorflow backend
+        with tempfile.NamedTemporaryFile(delete=False, suffix=".csv") as fd_out_tf:
+            deletes.append(fd_out_tf.name)
+            tf_args = args + ["--out", fd_out_tf.name, "--backend", "tensorflow"]
+            print("Running tensorflow with args: %s" % tf_args)
+            predict_command.run(tf_args)
+            fd_out_tf.close()  # Explicitly close file
+            result_tf = pandas.read_csv(fd_out_tf.name)
+            print("TensorFlow results:")
+            print(result_tf)
+
+        # Run with pytorch backend
+        with tempfile.NamedTemporaryFile(delete=False, suffix=".csv") as fd_out_torch:
+            deletes.append(fd_out_torch.name)
+            torch_args = args + ["--out", fd_out_torch.name, "--backend", "pytorch"]
+            print("Running pytorch with args: %s" % torch_args)
+            predict_command.run(torch_args)
+            fd_out_torch.close()  # Explicitly close file
+            result_torch = pandas.read_csv(fd_out_torch.name)
+            print("PyTorch results:")
+            print(result_torch)
+
+    finally:
+        # Make sure we've closed the files and pandas has released them
+        import gc
+        gc.collect()  # Force garbage collection
+        
+        # Add a small delay to ensure files are released
+        import time
+        time.sleep(0.1)
```

Comment:
This gc.collect and sleep call seems a bit hokey and I don't really follow the logic of the change. Is there a cleaner way?

Reply:
Will fix, thanks.

Reply:
Fixed.